src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1383,16 +1386,96 @@ AddUnionAllSetOperationsToAttributeEquivalenceClass(AttributeEquivalenceClass **
 			continue;
 		}
 		int rtoffset = RangeTableOffsetCompat(root, appendRelInfo);
+		int childRelId = appendRelInfo->child_relid - rtoffset;
 
-		/* set the varno accordingly for this specific child */
-		varToBeAdded->varno = appendRelInfo->child_relid - rtoffset;
+		RangeTblEntry *rte = root->simple_rte_array[childRelId];
```

Comment:
> Do you think it makes sense to error here if childRelId < root->simple_rel_array_size

Maybe not error, but just skip is better?

Reply:
umm, skipping might be better. But if that happens we would also want to know that. So how can we know that this happened in a production cluster without the user seeing an error etc? I guess this isn't possible

Reply:
we may have a DEBUG2 message?

Reply:
but DEBUG2 will not be logged in production in general?

I think you can choose either way (Skipping is already a good idea). My thinking was this "if there is a bug here in production, it will be hard to find. Hence if we could at least log this or somehow know that this happened, we can try to fix it without users realizing it"